source/cpu/security_amd64.go

```diff
@@ -102,6 +106,17 @@ func tdxEnabled() bool {
 	return false
 }
 
+func tdxProtected() bool {
+	features := cpuid.CPU.FeatureSet()
+	for _, feature := range features {
+		if feature == "TDX_GUEST" {
+			return true
+		}
+	}
```

Comment:
I think you could use `cpuid.CPU.Supports` instead?

Also, afaiu with the new cpuid we will also get `cpu-cpuid.TDX_GUEST` label, right? If we want this separate tdx label, I'd suggest to add that cpuid.TDX_GUEST label to the default deny list in `newDefaultConfig()` to avoid redundant duplicate information (i.e. excess labels). In this case we also need to update the documentation, and preferably the example config, regarding the default deny list.

Reply:
@marquiz, thanks for both suggestions.

About `cpuid.CPU.Supporst` (or `cpuid.CPU.Has`, I'll do the change right now.

About exposing the `cpuid.TDX_GUEST` label, I'd like to confirm one thing with you.  The main reason I do believe we should have it as `tdx.protected` instead of exposing `cpuid.TDX_GUEST` is for the sake of consistency.  You're 100% correct on  not exposing the `cpuid.TDX_GUEST` in this case, and I'd like to hear whether you agree with my consistency argument in order to expose `tdx.protected`.

Reply:
I think that consistency argument makes sense and I'm good with `tdx.protected`

Reply:
and yes, `cpuid.CPU.Has()` is better...